staker-rewards/lib/rewards-calculator.js

```diff
@@ -0,0 +1,116 @@
+import { callWithRetry } from "./contract-helper.js"
+
+export default class RewardsCalculator {
+  constructor(context, interval, minimumStake) {
+    this.context = context
+    this.interval = interval
+    this.ethScoreThreshold = 3000
+  }
+
+  static async initialize(context, interval, operatorsParameters) {
+    const rewardsCalculator = new RewardsCalculator(context, interval)
+
+    await rewardsCalculator.calculateOperatorsRewards(operatorsParameters)
+
+    return rewardsCalculator
+  }
+
+  async calculateOperatorsRewards(operatorsParameters) {
+    const minimumStake = await this.getMinimumStake()
+
+    const operatorsRewardsFactors = []
+
+    for (const operatorParameters of operatorsParameters) {
+      const { keepStaked, ethTotal } = operatorParameters.operatorAssets
+
+      const ethScore = this.calculateETHScore(ethTotal)
+      const boost = this.calculateBoost(keepStaked, ethTotal, minimumStake)
+      const rewardWeight = ethScore * boost
```

Comment:
Should we use multiplication from BN? Same about all other operations in this function.